src/pages/Analysis.tsx

```diff
@@ -0,0 +1,244 @@
+import { useEffect, useState, useCallback, useRef } from "react";
+import { useHackRFDevice } from "../hooks/useHackRFDevice";
+import InteractiveDSPPipeline from "../components/InteractiveDSPPipeline";
+import PerformanceMetrics from "../components/PerformanceMetrics";
+import Card from "../components/Card";
+import { Sample } from "../utils/dsp";
+import { performanceMonitor } from "../utils/performanceMonitor";
+import { ISDRDevice } from "../models/SDRDevice";
+
+const MAX_BUFFER_SAMPLES = 32768;
+const UPDATE_INTERVAL_MS = 33; // Target 30 FPS
+
+function Analysis(): React.JSX.Element {
+  const { device } = useHackRFDevice();
+  const [listening, setListening] = useState(false);
+  const [samples, setSamples] = useState<Sample[]>([]);
+  const [currentFPS, setCurrentFPS] = useState<number>(0);
+
+  const sampleBufferRef = useRef<Sample[]>([]);
+  const rafIdRef = useRef<number | null>(null);
+  const lastUpdateRef = useRef<number>(0);
+  const frameCountRef = useRef<number>(0);
+  const fpsLastUpdateRef = useRef<number>(0);
+  const receivePromiseRef = useRef<Promise<void> | null>(null);
```

Comment:
We're using useRef a lot; is there a better way to be handling state management?